### PR TITLE
Jdbc analyzers extend histogram

### PR DESCRIPTION
**Pull request by westphal-jan**
*Issue #52*

*Description of changes:*
Use a binning function when the `resultSet` is evaluated.
Add a function that returns the top `n` key-value pairs of a map with respect to the value without sorting of the pairs.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/main/scala/com/amazon/deequ/analyzers/jdbc/JdbcHistogram.scala [hunk 1]
```diff
@@ -14,23 +14,30 @@
...
+  private[this] val PARAM_CHECK: Table => Unit = { _ =>
+    if (maxDetailBins > JdbcHistogram.MaximumAllowedDetailBins) {
+      throw new IllegalAnalyzerParameterException(s"Cannot return histogram values for more " +
+        s"than ${JdbcHistogram.MaximumAllowedDetailBins} values")
+    }
+  }
+
   override def preconditions: Seq[Table => Unit] = {
-    hasTable() :: hasColumn(column) :: Nil
+    PARAM_CHECK :: hasTable() :: hasColumn(column) :: Nil
```
Comment (velramiir) [inline src/main/scala/com/amazon/deequ/analyzers/jdbc/JdbcHistogram.scala]: I'd suggest to stick to the precondition naming pattern here. All caps is usually used for constants, not for functions. Maybe name it "hasValidParams" or "hasValidMaxDetailParam"

**Review by velramiir — CHANGES_REQUESTED** [review]

File: src/main/scala/com/amazon/deequ/analyzers/jdbc/JdbcHistogram.scala (hunk 1 above)
Comment (velramiir) [reply src/main/scala/com/amazon/deequ/analyzers/jdbc/JdbcHistogram.scala]: So, Amazon did this, too. They named it PARAM_CHECKS so just add an 's'

Comment (velramiir) [inline src/main/scala/com/amazon/deequ/analyzers/jdbc/JdbcHistogram.scala]:
```suggestion
    PARAM_CHECKS :: hasTable() :: hasColumn(column) :: Nil
```

**Review by velramiir — DISMISSED** [review]

File: src/main/scala/com/amazon/deequ/analyzers/jdbc/JdbcHistogram.scala (hunk 1 above)
Comment (velramiir) [inline src/main/scala/com/amazon/deequ/analyzers/jdbc/JdbcHistogram.scala]:
```suggestion
    PARAM_CHECKS :: hasTable() :: hasColumn(column) :: Nil
```

File: src/main/scala/com/amazon/deequ/analyzers/jdbc/JdbcHistogram.scala [hunk 2]
```diff
@@ -14,23 +14,30 @@
...
 
 import scala.util.{Failure, Try}
 
-case class JdbcHistogram(column: String)
+case class JdbcHistogram(column: String,
+                         binningFunc: Option[Any => Any] = None,
+                         maxDetailBins: Integer = JdbcHistogram.MaximumAllowedDetailBins)
   extends JdbcAnalyzer[JdbcFrequenciesAndNumRows, HistogramMetric] {
 
+  private[this] val PARAM_CHECK: Table => Unit = { _ =>
```
Comment (velramiir) [inline src/main/scala/com/amazon/deequ/analyzers/jdbc/JdbcHistogram.scala]:
```suggestion
  private[this] val PARAM_CHECKS: Table => Unit = { _ =>
```

**Review by mschroederi — APPROVED** [review]

**Pull request merged**